app/Http/Controllers/LocationsController.php

```diff
@@ -210,5 +210,29 @@ public function show($id = null)
 
         return redirect()->route('locations.index')->with('error', trans('admin/locations/message.does_not_exist'));
     }
+    
+public function print_assigned($id)
+    {
+
+        $location = Location::where('id',$id)->first();
+        $parent = Location::where('id',$location->parent_id)->first();
+	$manager = User::where('id',$location->manager_id)->first();
+	$users = User::where('location_id', $id)->with('company', 'department', 'location')->get();
+        $assets = Asset::where('assigned_to', $id)->where('assigned_type', Location::class)->with('model', 'model.category')->get();
+        return view('locations/print')->with('assets', $assets)->with('users',$users)->with('location', $location)->with('parent', $parent)->with('manager', $manager);
+
+    }
+    
+    public function print_all_assigned($id)
+    {
+
+        $location = Location::where('id',$id)->first();
+        $parent = Location::where('id',$location->parent_id)->first();
+        $manager = User::where('id',$location->manager_id)->first();
+        $users = User::where('location_id', $id)->with('company', 'department', 'location')->get();
+        $assets = Asset::where('location_id', $id)->with('model', 'model.category')->get();
```

Comment:
This one seems very, very similar to the above, but I think the `location_id` clause in the `where()` might not be right - we do have an `rtd_location_id` for assets, but that's not what we're asking for. Or am I conflating those things, do we de-norm the location into `location_id` when an asset is checked out to a user? Not sure. Either way, these two routines look awfully similar and should probably be just one, or if not, should at least have an explanation as to why they're separate and what the difference might be.